docs/plugin/execution-modes.md

```diff
@@ -0,0 +1,117 @@
+* Resource control - the size of the thread pool and process priority can be set for the WPEProcess host, allowing more custom tuning for specific plugin requirements
+
+**Disadvantages**
+
+* Performance - since there is now an additional RPC hop over COM-RPC to invoke methods on the plugin, this can introduce some latency (although COM-RPC is efficient so this is very minimal)
+    * It may also take slightly longer to activate the plugin due to the overhead of starting the WPEProcess host
+* Resource usage - potentially increased resource usage from spawning and running a new process
+
+## Container
+
+!!! note
+	Support for containers is not enabled by default.
+	
+	To enable, build Thunder with the `-DPROCESSCONTAINERS=ON` cmake option and select a backend with `-DPROCESSCONTAINERS_XXX=ON` where `XXX` is the backend you wish to use.
+
+An extension to the out-of-process mode, container mode will run the WPEProcess host instead a containerised environment.  Thunder supports various container integrations using the `ProcessContainer` abstraction mechanism in Thunder core:
+
+- [LXC ](https://linuxcontainers.org/)
+- [runc](https://github.com/opencontainers/runc)
+- [crun](https://github.com/containers/crun)
+- [Dobby](https://github.com/rdkcentral/Dobby) (RDK)
+- AWC (Liberty Global)
+
+A requirement for running a plugin in a container is a suitable container configuration must have been defined - Thunder cannot create container configurations dynamically on-the-fly for plugins. Thunder will look for container configurations in the following locations (with the below priority):
+
+1. `<volatile path>/<callsign>Container`
+1. `<persistent path>/<callsign>/Container`
+1. `<data path>/<classname>/Container`
+
+For example, the config for a plugin will callsign SamplePlugin might be stored in `/opt/wpeframework/SamplePlugin/Container` if the persistent path is set to `/opt/wpeframework` in the Thunder configuration.
+
+**Advantages**
+
+* All the previous advantages of OOP plugins
+* Increased security - containers have restricted access to the host system including filesystem and device access
+* Improved resource limiting - containers can use cgroups to tightly control and monitor resource usage
+
+**Disadvantages**
+
+* Maintenance - need to maintain a container config and correctly punch holes in the container to access resources (e.g. device nodes)
+* Startup time - plugin activation time may be increased due to the overhead of constructing the container environment
+
+## Distributed
+
+!!! danger
+	Distributed mode is considered experimental
+
+Distributed mode takes the out-of-process mode one step further by allowing plugins to run on an entirely different device than the main WPEFramework process. This device could even be running a different CPU and with a architecture. A COM-RPC channel is established over a TCP socket between the two devices to allow communication.
```

Comment:
"This device could even be running a different CPU and with a architecture" -> "This device could even be running a different CPU with a different architecture"